c_binding/libsg.c

```diff
@@ -331,6 +338,59 @@ bool _sg_is_vpd_page_supported(uint8_t *vpd_0_data, uint8_t page_code)
     return false;
 }
 
+int _sg_parse_vpd_80(char *err_msg, uint8_t *vpd_data, uint8_t *serial_num,
+                     uint16_t serial_num_max_len)
+{
+    int rc = LSM_ERR_OK;
+    int len = 0;
+    struct _sg_t10_vpd80_header *vpd80_header = NULL;
+    uint8_t *p = NULL;
+    uint8_t *end_p = NULL;
+    uint16_t vpd80_len = 0;
+    uint16_t serial_num_len = 0;
+
+    assert(err_msg != NULL);
+    assert(vpd_data != NULL);
+    assert(serial_num != NULL);
+    assert(serial_num_max_len != 0);
+
+    memset(serial_num, 0, serial_num_max_len);
+
+    vpd80_header = (struct _sg_t10_vpd80_header*) vpd_data;
+
+    if (vpd80_header->page_code != _SG_T10_SPC_VPD_UNIT_SN) {
+        rc = LSM_ERR_LIB_BUG;
+        _lsm_err_msg_set(err_msg, "BUG: Got incorrect VPD page code '%02x', "
+                         "should be 0x80", vpd80_header->page_code);
+        goto out;
+    }
+
+    serial_num_len = be16toh(vpd80_header->page_len_be);
+    vpd80_len = serial_num_len + sizeof(struct _sg_t10_vpd83_header);
+
+    end_p = vpd_data + vpd80_len - 1;
+
+    if (end_p >= vpd_data + _SG_T10_SPC_VPD_MAX_LEN) {
+        rc = LSM_ERR_LIB_BUG;
+        _lsm_err_msg_set(err_msg, "BUG: Got invalid VPD UNIT SN page response, "
+                         "data length exceeded the maximum size of a legal VPD "
+                         "page");
+        goto out;
+    }
+
+    p = vpd_data + sizeof(struct _sg_t10_vpd83_header);
+
+    len = snprintf(serial_num, serial_num_len, "%s", p);
```

Comment:
Compile warning, should be:

`len = snprintf(serial_num, serial_num_len, "%s", (char *) p);`